src/main/kotlin/io/csbroker/apiserver/service/AuthServiceImpl.kt

```diff
@@ -35,15 +35,14 @@ class AuthServiceImpl(
     override fun saveUser(userDto: UserSignUpDto): UUID {
         val findUser = userRepository.findByEmailOrUsername(userDto.email, userDto.username)
 
-        if (findUser != null) {
-            if (findUser.email == userDto.email) {
+        findUser?.let {
+            if (it.email == userDto.email) {
                 throw ConditionConflictException(ErrorCode.EMAIL_DUPLICATED, "${userDto.email}은 중복 이메일입니다.")
-            } else {
-                throw ConditionConflictException(
-                    ErrorCode.USERNAME_DUPLICATED,
-                    "${userDto.username}은 중복 이메일입니다."
-                )
             }
+            throw ConditionConflictException(
+                ErrorCode.USERNAME_DUPLICATED,
+                "${userDto.username}은 중복 닉네임입니다.",
+            )
         }
```

Comment:
기존의 유저가 존재하는지 확인하는 validation 로직도 메서드 추출을 하면 가독성이 더 좋아질 것 같아요!!
혹은 findUser는 검증용으로만 쓰이니 변수에 담지 않고 바로 사용하는 것도 좋을 것 같네요 ‼️
```suggestion
    userRepository.findByEmailOrUsername(userDto.email, userDto.username)?.takeIf {
            if (it.email == userDto.email) {
                throw ConditionConflictException(ErrorCode.EMAIL_DUPLICATED, "${userDto.email}은 중복 이메일입니다.")
            }
            throw ConditionConflictException(
                ErrorCode.USERNAME_DUPLICATED,
                "${userDto.username}은 중복 닉네임입니다.",
            )
        }

Reply:
반영햇습니다.